core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -201,17 +201,20 @@ public void head(String path, Consumer<ErrorResponse> errorHandler) {
   }
 
   @Override
-  public <T> T get(String path, Class<T> responseType, Consumer<ErrorResponse> errorHandler) {
+  public <T extends RESTResponse> T get(String path, Class<T> responseType,
+                                        Consumer<ErrorResponse> errorHandler) {
     return execute(Method.GET, path, null, responseType, errorHandler);
   }
 
   @Override
-  public <T> T post(String path, Object body, Class<T> responseType, Consumer<ErrorResponse> errorHandler) {
+  public <T extends RESTResponse> T post(String path, Object body, Class<T> responseType,
```

Comment:
Should body be something that extends `RESTRequest`?

Reply:
Yeah. I used just `RESTRequest` here, but not certain if it needs to be another type variable here for post of like `<Req extends RESTRequst>`.

I think just using `RESTRequest` is fine.